py-rse/errors.Rmd

```diff
@@ -564,23 +566,37 @@ because then they can set it up the way they want with very little work.
 
 ## Exercises {#py-rse-errors-exercises}
 
-### Set the logging level {#py-rse-errors-ex-set-level}
+### Handling exceptions {#py-rse-errors-ex-exceptions}
 
--   FIXME: use `getopt` and `--level NAME` to set logging level.
+1.  Modify the script `collate.py` so that it catches any exceptions
+    that are raised when it tries to open files.
+    When you are finished,
+    the program should collate all the files it can
+    rather than halting as soon as it encounters a problem.
+2.  Modify your first solution to handle nonexistent files
+    and permission problems separately.
 
-### Display logging configuration {#py-rse-errors-ex-display-log-config}
+### Error catalogs {#py-rse-errors-ex-catalog}
 
--   FIXME: use `logging_tree.printout()` to display logging configuration.
+1.  Modify your solution to the previous exercise
+    to put your error message in a catalog
+    as described in Section \@ref(py-rse-errors-messages).
+2.  Add messages in a second language.
+    (Use Google Translate if necessary).
+3.  Add a command-line flag to allow users to select the language they want to use.
 
-### Log to standard error as well as a file {#py-rse-errors-ex-log-stderr}
+### Set the logging level {#py-rse-errors-ex-set-level}
```

Comment:
How does this exercise fit with the fact that in #343 logging is now added to `collate.py` during the lesson?